spec/uploaders/logo_uploader_spec.rb

```diff
@@ -0,0 +1,97 @@
+require "rails_helper"
+require "carrierwave/test/matchers"
+require "exifr/jpeg"
+
+describe LogoUploader, type: :uploader do
+  include CarrierWave::Test::Matchers
+
+  let(:image_svg) { fixture_file_upload("300x100.svg") }
+  let(:image_jpg) { fixture_file_upload("800x600.jpg", "image/jpeg") }
+  let(:image_png) { fixture_file_upload("800x600.png", "image/png") }
+  let(:image_webp) { fixture_file_upload("800x600.webp", "image/webp") }
+  let(:image_with_gps) { fixture_file_upload("image_gps_data.jpg", "image/jpeg") }
+  let(:image_gif) { fixture_file_upload("high_frame_count.gif", "image/gif") }
+
+  # we need a new uploader before each test, and since the uploader is not a model
+  # we can recreate it quickly in memory with `let!`
+  let!(:uploader) { described_class.new }
+
+  before do
+    described_class.include CarrierWave::MiniMagick # needed for processing
+    described_class.enable_processing = true
+  end
+
+  after do
+    described_class.enable_processing = false
+    uploader.remove!
+  end
+
+  it "stores files in the correct directory" do
+    expect(uploader.store_dir).to eq("uploads/logos/")
+  end
+
+  describe "formats" do
+    it "permits a set of extensions" do
+      expect(uploader.extension_allowlist).to eq(%w[svg png jpg jpeg jpe])
+    end
+
+    it "permits jpegs" do
+      uploader.store!(image_jpg)
+      expect(uploader).to be_format("jpeg")
+    end
+
+    it "permits pngs" do
+      uploader.store!(image_png)
+      expect(uploader).to be_format("png")
+    end
+
+    it "permits svgs" do
+      uploader.store!(image_svg)
+      expect(uploader).to be_format("svg")
+    end
+
+    it "rejects unsupported formats like webp" do
+      expect { uploader.store!(image_webp) }.to raise_error(CarrierWave::IntegrityError)
+    end
+
+    it "rejects unsupported formats like gifs" do
+      expect { uploader.store!(image_gif) }.to raise_error(CarrierWave::IntegrityError)
+    end
+  end
+
+  describe "error handling" do
+    it "raises a CarrierWave error which can be parsed if MiniMagick timeout occurs" do
+      allow(MiniMagick::Image).to receive(:new).and_raise(Timeout::Error)
+
+      expect { uploader.store!(image_jpg) }.to raise_error(CarrierWave::IntegrityError, /Image processing timed out/)
+    end
+  end
+
+  describe "exif removal" do
+    it "removes EXIF and GPS data on single frame image upload" do
+      expect(EXIFR::JPEG.new(image_with_gps.path).exif?).to be(true)
+      expect(EXIFR::JPEG.new(image_with_gps.path).gps.present?).to be(true)
+      uploader.store!(image_with_gps)
+      expect(EXIFR::JPEG.new(uploader.file.path).exif?).to be(false)
+      expect(EXIFR::JPEG.new(uploader.file.path).gps.present?).to be(false)
+    end
+  end
+
+  describe "resize_image" do
+    it "creates versions of the image with different filenames" do
```

Comment:
Same feedback as my previous comment!

**_Quick, non-blocking suggestion_**: since we have a handful of expectations here, using [`:aggregate_failures`](https://www.rubydoc.info/gems/rspec-expectations/RSpec%2FMatchers:aggregate_failures) here could be beneficial:
```suggestion
    it "creates versions of the image with different filenames", :aggregate_failures do
```

Reply:
✅